### PR TITLE
[SPARK-26508][Core][SQL] Address warning messages in Java reported at lgtm.com

### PR DESCRIPTION
**Pull request by kiszk**
## What changes were proposed in this pull request?

This PR addresses warning messages in Java files reported at [lgtm.com](https://lgtm.com).

[lgtm.com](https://lgtm.com) provides automated code review of Java/Python/JavaScript files for OSS projects. [Here](https://lgtm.com/projects/g/apache/spark/alerts/?mode=list&severity=warning) are warning messages regarding Apache Spark project.

This PR addresses the following warnings:

- Result of multiplication cast to wider type
- Implicit narrowing conversion in compound assignment
- Boxed variable is never null
- Useless null check

NOTE: `Potential input resource leak` looks false positive for now.

## How was this patch tested?

Existing UTs

### CODE REVIEW

File: common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public double toBytes(long d) {
     if (d < 0) {
       throw new IllegalArgumentException("Negative size value. Size must be positive: " + d);
     }
-    return d * multiplier;
+    return d * (double)multiplier;
```
Comment (srowen) [inline common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java]: We need a space after casts. I suppose the point is to avoid overflow when the result is going to be a double anyway, at the cost of precision maybe. I tested, and the precision issue won't matter, but then again, there's no reason this method should return `double`. There is no such thing as 0.5 bytes.

I think we can just fix that for Spark 3; its value is used as a `long` or `int` everywhere anyway.

While here, I think I'd also fix the expressions like `(long) Math.pow(1024L, 2L)` above. The multipliers should be `1L << 10`, `1L << 20`, ... `1L << 50`. There's again no point there in using floating point math.

File: common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java [hunk 2]
```diff
@@ -311,6 +311,6 @@ public int chunkFetchHandlerThreads() {
       conf.getInt("spark.shuffle.server.chunkFetchHandlerThreadsPercent", 100);
     return (int)Math.ceil(
      (this.serverThreads() > 0 ? this.serverThreads() : 2 * NettyRuntime.availableProcessors()) *
-     chunkFetchHandlerThreadsPercent/(double)100);
+     (double)chunkFetchHandlerThreadsPercent/100);
```
Comment (srowen) [inline common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java]: Write 100.0 here for extra clarity, and put spaces after all the casts here. Otherwise, what's the issue with this expression?

File: core/src/main/scala/org/apache/spark/package.scala [hunk 3]
```diff
@@ -76,7 +76,9 @@ package object spark {
         )
       } catch {
         case e: Exception =>
-          throw new SparkException("Error loading properties from spark-version-info.properties", e)
+      //  throw new SparkException("Error loading properties from spark-version-info.properties", e)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/package.scala]: What's the purpose of this change?

File: core/src/main/scala/org/apache/spark/package.scala [hunk 4]
```diff
@@ -76,7 +76,9 @@ package object spark {
...
+          val unknownProp = "<unknown>"
+          (unknownProp, unknownProp, unknownProp, unknownProp, unknownProp, unknownProp)
```
Comment (viirya) [inline core/src/main/scala/org/apache/spark/package.scala]: Is this safe change? Looks like behavior change when throwing exception.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/xml/UDFXPathUtil.java [hunk 5]
```diff
@@ -182,7 +182,7 @@ public long skip(long ns) throws IOException {
       // Bound skip by beginning and end of the source
       long n = Math.min(length - next, ns);
       n = Math.max(-next, n);
-      next += n;
+      next += (int)n;
```
Comment (srowen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/xml/UDFXPathUtil.java]: I'd put the cast on line 183 and make n an int. `length` and `next` are nonnegative ints.

File: launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java [hunk 6]
```diff
@@ -320,7 +320,7 @@ protected void handle(Message msg) throws IOException {
         } else {
           if (handle == null) {
             throw new IllegalArgumentException("Expected hello, got: " +
-              msg != null ? msg.getClass().getName() : null);
+              (msg != null ? msg.getClass().getName() : "null"));
```
Comment (srowen) [inline launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java]: You could probably just improve this by creating this class name string once above line 321, and putting something like "no message" instead of null, while we're here.

**Review by srowen — CHANGES_REQUESTED** [review]

File: core/src/main/scala/org/apache/spark/package.scala (hunk 4 above)
Comment (kiszk) [reply core/src/main/scala/org/apache/spark/package.scala]: Thank you, this is my mistake. I will revert this.

File: core/src/main/scala/org/apache/spark/package.scala (hunk 3 above)
Comment (kiszk) [reply core/src/main/scala/org/apache/spark/package.scala]: ditto, I will revert this.

File: common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java (hunk 2 above)
Comment (kiszk) [reply common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java]: `lgtm.com` and I are afraind `(...) * chunkFetchHandlerThreadsPercent` may cause overflow since it is `int * int`. After this change, it will be `double * double`.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/xml/UDFXPathUtil.java (hunk 5 above)
Comment (kiszk) [reply sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/xml/UDFXPathUtil.java]: i see

File: launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java (hunk 6 above)
Comment (kiszk) [reply launcher/src/main/java/org/apache/spark/launcher/LauncherServer.java]: Sure

File: common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java (hunk 2 above)
Comment (srowen) [reply common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java]: Oh it's because it's really (threads * percent) / 100. Maybe this can be simplified by breaking out the `(this.serverThreads() ...)` into a local variable, then writing as threads * (percent / 100.0)

**Comment by kiszk** [comment]
not closed this

**Comment by kiszk** [comment]
reopen this

**Comment by kiszk** [comment]
retest this please

File: common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java (hunk 1 above)
Comment (kiszk) [reply common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java]: I realized that we need `return double` since `return long` causes a failure at [here](https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/util/UtilsSuite.scala#L229) regarding terabyte.

Comment (srowen) [reply common/network-common/src/main/java/org/apache/spark/network/util/ByteUnit.java]: Hm, really? what does it return? type-wise the numbers should easily fit into a long. 2 TiB = 2199023255552 bytes. I'm missing where the failure is here.

**Review by holdenk — COMMENTED** [review]
I'm not sure about the cast with the division, could we simplify it?

**Comment by kiszk** [comment]
retest this please

**Review by srowen — CHANGES_REQUESTED** [review]
Looks fine pending tests, with one tiny request

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master
